experimental/PropertyDDS/packages/property-dds/src/test/propertyTree.spec.ts

```diff
@@ -195,6 +172,100 @@ describe("PropertyDDS summarizer", () => {
 
 		expect(u1.root.get<ArrayProperty>(USERS)?.getValues().length).to.equal(1);
 	});
+
+	it("Scenario 2 (repeated summarization)", async function () {
+		/**
+		 * This test produces a scenario where we have an empty remoteChanges array in the summarizer client
+		 * and then get more changes that cannot yet be pruned away, because the MSN continues to point to the
+		 * previous head commit.
+		 *
+		 * We used to have a bug that was caused by this, where the prune code would prune the remote changes
+		 * but not the unrebased remote changes, causing rebase errors.
+		 */
+		this.timeout(30000);
+
+		// 1- U1 joins together with summarizer
+		const {
+			client: u1,
+			summarizer,
+			dataObject: dataObject1,
+			container: container1,
+		} = await getClient(true);
+		await objProvider.ensureSynchronized();
+
+		// 2- Make some modifications
+		u1.root.insert("c1", PropertyFactory.create("NodeProperty"));
+		u1.commit();
+		await objProvider.ensureSynchronized();
+
+		u1.root.insert("c2", PropertyFactory.create("NodeProperty"));
+		u1.commit();
+
+		await objProvider.ensureSynchronized();
+
+		const {
+			client: u2,
+			dataObject: dataObject2,
+			container: container2,
+		} = await getClient(false, true);
+		await objProvider.ensureSynchronized();
+
+		// We do two changes to a different DDS (the root map), to make sure, that
+		// updates are triggered that do not affect the propertyDDS
+		dataObject1.root.set("c2", "aaa");
+		dataObject2.root.set("c2", "aaa");
+		await objProvider.ensureSynchronized();
+
+		// Now we wait until the msn has sufficiently advanced that the pruning below
+		// will remove all remoteChanges
+		await synchronizeMSN(container2, container1);
+
+		// Summarize
+		await summarizeNow(summarizer.summarizer);
+
+		const summarizerDataObject =
+			(await summarizer.container.getEntryPoint()) as ITestFluidObject;
```

Comment:
I think this cast is wrong, since the line below is failing. This was based off the guidance in https://github.com/microsoft/FluidFramework/pull/18491#issuecomment-1828704544 but I think I'm missing something.

Reply:
I have applied a workaround from @kian-thompson and the test now passes.